### PR TITLE
[WIP] Add gotty

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is my first go package. Can someone review please?

Also: This builds fine, but **I do not get an executable** out of the build process. Do I have to tweak things somewhere?

### CODE REVIEW

File: pkgs/top-level/go-packages.nix [hunk 1]
```diff
@@ -796,6 +796,22 @@ let
     sha256 = "0yg1jpr7lcaqh6i8n9wbs9r128kk541qjv06r9a6fp9vj56rqr3m";
   };
 
+  gotty = buildFromGitHub {
+    rev     = "8f9d5ba582e4ad5e4ef41fe941cdd6280917aff1";
```
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: Latest release is v0.0.10

File: pkgs/top-level/go-packages.nix [hunk 2]
```diff
@@ -989,6 +1005,20 @@ let
     };
   };
 
+  go-bindata-assetfs = buildFromGitHub {
```
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: Please add a date since there is no release

File: pkgs/top-level/go-packages.nix [hunk 3]
```diff
@@ -1676,6 +1706,20 @@ let
     '';
   };
 
+  manners = buildFromGitHub {
+    rev = "0b5e6b2c2843f4c83c2a40f96980b09cf4af733c";
```
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: Latest release is 0.4.0

File: pkgs/top-level/go-packages.nix [hunk 4]
```diff
@@ -989,6 +1005,20 @@ let
...
+    rev     = "d5cac425555ca5cf00694df246e04f05e6a55150";
+    owner   = "elazarl";
+    repo    = "go-bindata-assetfs";
+    sha256  = "636ce247ff6f85c14f38a421f46662fa77bdc29762692e1f72b3cd1f9d7a1d17";
+
+    meta = {
+      description = "Serves embedded files from jteeuwen/go-bindata with net/http";
+      homepage = "https://github.com/elazarl/go-bindata-assetfs";
+      maintainers = with lib.maintainers; [ matthiasbeyer ];
+      license = licenses.bsd;
```
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: There is no such license, the license is the repo is the bsd2 one

**Comment by edwtjo** [comment]
You're missing a

```
    postInstall = ''
      mkdir $out/bin
      cp go/bin/gotty $out/bin/gotty
    '';
```
In your gotty expression. You might want to recheck the others as well.

File: pkgs/top-level/go-packages.nix (hunk 1 above)
Comment (matthiasbeyer) [inline pkgs/top-level/go-packages.nix]: You mean I can specify a tag name in the rev field (This rev is from v0.0.10)?

File: pkgs/top-level/go-packages.nix (hunk 2 above)
Comment (matthiasbeyer) [inline pkgs/top-level/go-packages.nix]: As comment?

File: pkgs/top-level/go-packages.nix (hunk 1 above)
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: Yes exactly, fetchFromGithub is the underlying fetcher and the archive API call is the same for revs/tags.

File: pkgs/top-level/go-packages.nix (hunk 2 above)
Comment (edwtjo) [inline pkgs/top-level/go-packages.nix]: As an attribute `date = ... ;`

File: pkgs/top-level/go-packages.nix (hunk 1 above)
Comment (matthiasbeyer) [inline pkgs/top-level/go-packages.nix]: TIL. Thankyou!

**Comment by matthiasbeyer** [comment]
Okay, it builds for me now, I can start it, but only with the storage path. The last output of the build process is this:

```
patching script interpreter paths in /nix/store/z0180xxhqyw82dlwiy79mzk514j3fdv6-go1.5-gotty-v0.0.10-bin
output ‘/nix/store/apwnn7vvcr4jb28y49sbji9xnlrqg81m-go1.5-gotty-v0.0.10’ is not allowed to refer to path ‘/nix/store/f8wvc00zgjlgrsh6wj7v1nbjp6vr7k25-go-1.5.1’
error: build of ‘/nix/store/sdmg98lqfhdfqsa8gl9d9inxfh92plji-go1.5-gotty-v0.0.10.drv’ failed
```

File: pkgs/top-level/go-packages.nix (hunk 2 above)
Comment (matthiasbeyer) [inline pkgs/top-level/go-packages.nix]: Done. Diff does not outdate.

**Comment by edwtjo** [comment]
That's wierd, someone with more intimate knowledge about the go builder will have to look at that. Perhaps @lethalman ?

**Comment by matthiasbeyer** [comment]
@edwtjo @lethalman The binary `${gotty}/bin/gotty` refers to `/nix/store/f8wvc00zgjlgrsh6wj7v1nbjp6vr7k25-go-1.5.1` which is obviousely wrong. I don't know how to change the build to fix this, though.

**Comment by edwtjo** [comment]
Merged with 09895a3f7111affc4d1b9cf2469e8c2fbee96085 98b3626bf203a6e8867ce498105056722f194ed0 700c3c93fb26a60b3a11082323cb399212076a78

**Comment by matthiasbeyer** [comment]
The issue was that I `cp`'d the binary and the solution was to `ln` it? Can you explain?

**Comment by edwtjo** [comment]
It seemed that my initial suggestion to use `cp` wasn't very good, I just assumed that the go subdirectory was in a temporary build dir when in fact it was already in the store. This has probably something to do with the go builder which probably populates some sort of environment for building, I still haven't read up. In any case, none of that really matters since gotty was already in a safe store path, i.e. just linking it in bin presents the executable to the profile.

**Pull request closed without merging**